axelrod/strategies/grudger.py

```diff
@@ -342,3 +342,75 @@ def strategy(opponent: Player) -> Action:
         elif opponent.defections:
             return D
         return C
+
+
+class CAPRI(Player):
+    """
+    CAPRI is a memory-3 strategy proposed in [Murase2020]_. Its behavior is defined by the following five rules.
+    C: Cooperate at mutual cooperation.
+        This rule prescribes c at (ccc, ccc).
+    A: Accept punishment when you mistakenly defected from mutual cooperation.
+        This rule prescribes c at (ccd, ccc), (cdc, ccd), (dcc, cdc), and (ccc, dcc).
+    P: Punish your co-player by defecting once when he defected from mutual cooperation.
+        This rule prescribes d at (ccc, ccd), and then c at (ccd, cdc), (cdc, dcc), and (dcc, ccc).
+    R: Recover cooperation when you or your co-player cooperated at mutual defection.
+        This rule prescribes c at (ddd, ddc), (ddc, dcc), (dcc, ccc), (ddc, ddd), (dcc, ddc), (ccc, dcc), (ddc, ddc), and (dcc, dcc).
+    I: In all the ohter cases, defect.
```

Comment:
These rules are not immediately clear to me. If it is verbatim from the paper (which I have not read yet) then that should be made clear and some clarification should be added.

Reply:
(The rules are from the paper)